runtime/caml/io.h

```diff
@@ -103,6 +103,11 @@ CAMLextern intnat caml_really_getblock (struct channel *, char *, intnat);
 CAMLextern void caml_channel_lock(struct channel *);
 CAMLextern void caml_channel_unlock(struct channel *);
 
+/* Lock and Unlock are compatibility aliases for OCaml<5.2.
+   Remove whenever 5.2 is old enough. (See #12792.) */
```

Comment:
For me, it's so much "old enough" as "whenever noone still targets both current OCaml and < 5.2" (and the deprecations in 5.0.0 suggest that if there's no reason not to support a wide range, code will........!). Perhaps:
```suggestion
/* Prior to OCaml 5.2, caml_channel_lock and caml_channel_unlock were hooks and
   Lock and Unlock provided a neater way of calling them in io.c. The macros are
   kept for third party code which manipulates channels in C, but targets both
   current OCaml and versions prior to 5.2 (see #12792.) */
```

Reply:
I was not fully confident that we need to be so detailed in our comment, given that "what's a good time to remove a compatibility alias" is a fairly generic question with always the same answer: "whenever no one really needs it anymore". I generally like to be detailed in comments, but I was too lazy to decide whether to integrate your comment or not, so I just went without as the path of least effort.